### PR TITLE
refactor helm runner

### PR DESCRIPTION
**Pull request by YaaraVerner**
extract from the run function the logic of parsing the output and converting helm to k8s.
create `K8sHelmRunner` class that creates helm report from given `chart_dir_and_meta`.
save the converting result in a property in the new class in order to allow the separation in the feature.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/helm/runner.py [hunk 1]
```diff
@@ -3,25 +3,70 @@
...
 
+class K8sHelmRunner(k8_runner):
+    def __init__(self, graph_class: Type[LocalGraph] = KubernetesLocalGraph,
+                 db_connector: NetworkxConnector = NetworkxConnector(),
+                 source: str = "Kubernetes",
+                 graph_manager: Optional[GraphManager] = None,
+                 external_registries: Optional[List[BaseRegistry]] = None) -> None:
+        super().__init__(graph_class, db_connector, source, graph_manager, external_registries)
+        self.chart_dir_and_meta = []
+        self.check_type = CheckType.HELM
```
Comment (gruebel) [inline checkov/helm/runner.py:35]: can you move the `check_type` attribute to the class level, like in other runners?

File: checkov/helm/runner.py [hunk 2]
```diff
@@ -3,25 +3,70 @@
...
+
+    def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=RunnerFilter(), collect_skip_comments=True, helmChart=None):
```
Comment (gruebel) [inline checkov/helm/runner.py:37]: can you add type hints here? 🙂

File: checkov/helm/runner.py [hunk 3]
```diff
@@ -3,25 +3,70 @@
...
+        if external_checks_dir:
+            for directory in external_checks_dir:
+                registry.load_external_checks(directory)
```
Comment (gruebel) [inline checkov/helm/runner.py:40]: move this after `if not self.chart_dir_and_meta: ...`, no. need to do it, if there are no charts 😄

File: checkov/helm/runner.py [hunk 4]
```diff
@@ -61,13 +110,12 @@ def parse_helm_dependency_output(o):
...
             lines = output.split('\n')
             for line in lines:
-                if line != "":
-                    if "NAME" not in line:
-                        chart_name, chart_version, chart_repo, chart_status = line.split("\t")
-                        chart_dependencies.update({chart_name.rstrip(): {'chart_name': chart_name.rstrip(),
-                                                                         'chart_version': chart_version.rstrip(),
-                                                                         'chart_repo': chart_repo.rstrip(),
-                                                                         'chart_status': chart_status.rstrip()}})
+                if line != "" and "NAME" not in line:
```
Comment (gruebel) [inline checkov/helm/runner.py:113]:
```suggestion
                if line and "NAME" not in line:
```

File: checkov/helm/runner.py [hunk 5]
```diff
@@ -98,138 +146,101 @@ def check_system_deps(self):
             logging.info(f"Error running necessary tools to process {self.check_type} checks.")
             return self.check_type
 
-    def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=RunnerFilter(),
-            collect_skip_comments=True):
+    def _parse_output(self, target_dir, output):
```
Comment (gruebel) [inline checkov/helm/runner.py:149]: type hints, please 😄

File: checkov/helm/runner.py [hunk 6]
```diff
@@ -3,25 +3,70 @@
...
+            try:
+                target_dir = os.path.join(root_folder, chart_dir)
+                chart_results = super().run(target_dir, external_checks_dir=external_checks_dir,
+                                            runner_filter=runner_filter, helmChart=chart_meta['name'])
+                fix_report_paths(chart_results, target_dir)
+                logging.debug(f"Sucessfully ran k8s scan on {chart_meta['name']}. Scan dir : {target_dir}")
+                report.failed_checks += chart_results.failed_checks
+                report.passed_checks += chart_results.passed_checks
+                report.parsing_errors += chart_results.parsing_errors
+                report.skipped_checks += chart_results.skipped_checks
```
Comment (tronxd) [inline checkov/helm/runner.py:54]:
```suggestion
                report.failed_checks.extend(chart_results.failed_checks)
                report.passed_checks.extend(chart_results.passed_checks)
                report.parsing_errors.extend(chart_results.parsing_errors)
                report.skipped_checks.extend(chart_results.skipped_checks)
```

File: checkov/helm/runner.py [hunk 7]
```diff
@@ -98,138 +146,101 @@ def check_system_deps(self):
...
+            except Exception:
+                logging.info(
+                    f"Error processing helm chart {chart_meta['name']} at dir: {chart_dir}. Working dir: {target_dir}.",
+                    exc_info=True,
+                )
+
+            self._parse_output(target_dir, o)
+            return chart_dir_and_meta
+
+    def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=RunnerFilter(),
```
Comment (gruebel) [inline checkov/helm/runner.py:239]: type hints 🙂

**Review by tronxd — APPROVED** [review]
LGTM

**Pull request merged**